packages/test/test-end-to-end-tests/src/test/SummarizerWithDataStoreRealization.spec.ts

```diff
@@ -0,0 +1,329 @@
+            // re-sent. Do it here so that the extra events don't mess with rest of the test.
+            mainDataStore = await requestFluidObject<TestDataObject1>(mainContainer, "default");
+            mainDataStore._root.set("anytest", "anyvalue");
+            await waitForContainerConnection(mainContainer);
+        });
+
+        it("No Summary Upload Error when DS gets realized between summarize and completeSummary", async () => {
+            const summarizerClient = await createSummarizer(provider, mainContainer);
+            await provider.ensureSynchronized();
+            mainDataStore.matrix.setCell(0, 0, "value");
+
+            // Here are the steps that would cause bug 1633 to repro:
+            // Additional information: https://dev.azure.com/fluidframework/internal/_workitems/edit/1633
+            // 1) Summary starts
+            // 2) The summarize method from the DataStore2 (TestDataObject2) will be executed but, as it has not
+            //    been realized, it has no child nodes and hasn't changed, we will use a handle instead.
+            // 3) During the summarization from the other DataStore1 (TestDataObject1),
+            // due to the mixinSummaryHandler (search) we explicitly realize the DataStore2 and
+            // new Summarizer Nodes are added to it.
+            // 4) That would (without the fix) corrupt the pendingSummaries/lastSummary from one of the child nodes.
+            // 5) Next Summarization starts, the lastSummary data would be used to upload the summary and we
+            //  would get an error
+            // "Cannot locate node with path '.app/.channels/guid1/root' under '<handle>'."
+            //  instead of .app/.channels/guid1/.channels/root
+
+            const summaryVersion = await waitForSummary(summarizerClient);
+            assert(summaryVersion, "Summary version should be defined");
+
+            mainDataStore.matrix.setCell(0, 0, "value1");
+            // The new summarization would immediately trigger bug 1633.
+            const summaryVersion1 = await waitForSummary(summarizerClient);
+            assert(summaryVersion1, "Summary version should be defined");
+
+            // Make sure the next summarization succeeds.
+            mainDataStore.matrix.setCell(0, 0, "value1");
+            const summaryVersion2 = await waitForSummary(summarizerClient);
+            assert(summaryVersion2, "Summary version should be defined");
+
+            summarizerClient.close();
+
+            // Just make sure new summarizer will be able to load and execute successfully.
+            const summarizerClient2 = await createSummarizer(provider, mainContainer,
+                summaryVersion2);
+
+            mainDataStore.matrix.setCell(0, 0, "value2");
+            const summaryVersion3 = await waitForSummary(summarizerClient2);
+            assert(summaryVersion3, "Summary version should be defined");
```

Comment:
In the failure case (with the bug), is this the step that fails?

Reply:
No, line 310 fails. This is just to make sure we are good after another summarizer takes over.

Reply:
The line numbers seem to have updated. You mean the one with the comment "The new summarization would immediately trigger bug 1633." will fail right.